### PR TITLE
Iox #15 add docker environment

### PR DESCRIPTION
**Pull request by Mr-Slippery**
This adds a Docker environment for building and running iceoryx applications as described in #15.

It suggests a usage pattern based on `docker exec` and one based on `screen`, depending on preference and configuration.

Further details are on the description of related issue #15 and inside the added documentation files.

Thanks for your kind feedback!

### CODE REVIEW

File: tools/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        libncurses5-dev \
+        pkg-config \
+        screen \
+        git
+
+ADD . /
+
+WORKDIR /iceoryx
+
+RUN ./tools/iceoryx_build_test.sh \
```
Comment (mossmaurice) [inline tools/docker/Dockerfile]: In your original version (https://github.com/Mr-Slippery/iceoryx-dockerfiles/blob/master/Dockerfile) you were cloning iceoryx before executing this script. How is the script coming in the docker container?

File: tools/docker/README.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+$ docker exec -it roudi /bin/bash
+root@03043f96ae10:/iceoryx# ./build/iceoryx_examples/icedelivery/ice_receiver
+Not subscribed
+Receiving: 78
+Receiving: 79
+Receiving: 80
+Receiving: 81
+```
+
+## Using `screen` to launch RouDi and examples
```
Comment (mossmaurice) [inline tools/docker/README.md]: Great chapter!

File: tools/docker/.screenrc [hunk 3]
```diff
@@ -0,0 +1 @@
+shell "/bin/bash"
```
Comment (mossmaurice) [inline tools/docker/.screenrc]: I'm quite fond of [fish](https://de.wikipedia.org/wiki/Fish_(Shell)). Could we use it here?

File: tools/docker/Dockerfile (hunk 1 above)
Comment (Mr-Slippery) [reply tools/docker/Dockerfile]: Hi @mossmaurice, indeed, initially I was cloning, but for this "integrated version", I thought it makes sense to build and include into the container the current source that is cloned. This makes it into the container via `ADD . /` in the Dockerfile.

Comment (Mr-Slippery) [reply tools/docker/Dockerfile]: But now that you mention it, there is perhaps an issue here, maybe some unintended things make it inside as well, I'll check and fix this.

Comment (Mr-Slippery) [reply tools/docker/Dockerfile]: Updated now, indeed, there was an issue, I was going to the parent of top-level `iceoryx` and adding everything into Docker `/`. Now I just go to `iceoryx` and add its content into Docker `/iceoryx`. Rebased branch, melded changes into existing commit. Thanks for the catch!

File: tools/docker/.screenrc (hunk 3 above)
Comment (Mr-Slippery) [reply tools/docker/.screenrc]: I don't see any problem, could install fish as extra package just like screen. I'll give it a try.

Comment (Mr-Slippery) [reply tools/docker/.screenrc]: Changed the screen shell to fish now (rebased/melded changes).

```
Welcome to fish, the friendly interactive shell
root@252a62dfbf2e /iceoryx# ./build/iceoryx_examples/icedelivery/ice_sender
Sending: 0
Sending: 1
Sending: 2
Sending: 3
```

File: tools/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2019 by Cornel Izbasa. All rights reserved.
```
Comment (dkroenke) [inline tools/docker/Dockerfile]: I guess this is you @Mr-Slippery? I need to confirm that we have the approval from the Author to use this codesnippet.

Comment (Mr-Slippery) [reply tools/docker/Dockerfile]: Hi @dkroenke, yes, it's just my Github username. :) The Eclipse account linked to this Github username has my real name, Cornel Izbasa. Thanks!

Comment (dkroenke) [reply tools/docker/Dockerfile]: Are you maybe located in Berlin? I found a LinkedIn Profile which says that you are in Tempelhof.

Comment (Mr-Slippery) [reply tools/docker/Dockerfile]: I work at HELLA in Tempelhof, but live in North-East of Berlin :)

**Comment by Mr-Slippery** [comment]
@dkroenke absolutely agree with the docker-compose follow-up, for starters I just didn't want to ask too much of the user, except to have Docker installed and then be able to build and run the apps. And if the image is published on Docker Hub then it will be even easier.

**Pull request merged**